src/prefect/engine.py

```diff
@@ -343,24 +343,22 @@ async def begin_flow_run(
             report_flow_run_crashes(flow_run=flow_run, client=client)
         )
 
-        # Create a task group for background tasks
-        flow_run_context.background_tasks = await stack.enter_async_context(
-            anyio.create_task_group()
+        flow_run_context.task_runner = await stack.enter_async_context(
+            flow.task_runner.start()
         )
 
         # If the flow is async, we need to provide a portal so sync tasks can run
-        flow_run_context.sync_portal = (
-            stack.enter_context(start_blocking_portal()) if flow.isasync else None
+        flow_run_context.sync_portal = stack.enter_context(start_blocking_portal())
```

Comment:
Always have sync portal up; this cleaned up logic downstream, but depending on whether running `start_blocking_portal` is expensive, add back https://github.com/PrefectHQ/prefect/pull/7041/commits/7c16f0298aa84a3ce9c5b03481c3003c226436b5